ql/src/java/org/apache/hadoop/hive/ql/exec/ReplCopyTask.java

```diff
@@ -177,6 +184,24 @@ protected int execute(DriverContext driverContext) {
         listBW.close();
       }
 
+      // If the srcMap is not empty which means we made the list of files for distCp. If there are files from different
+      // filesystems, then the map have multiple entries.
+      if (!srcMap.isEmpty()) {
+        boolean success = true;
+        for (final HashMap.Entry<FileSystem, List<Path>> entry : srcMap.entrySet()) {
+          FileSystem actualSrcFs = entry.getKey();
+          List<Path> srcPaths = entry.getValue();
+          if (!doCopy(toPath, dstFs, srcPaths, actualSrcFs, conf)) {
+            console.printError("Failed to copy: " + srcPaths.size()
+                    + " files to: '" + toPath.toString() + "'");
+            success = false;
```

Comment:
Here, doCopy fails if any file in the list doesn't exist which can happen only if the path refers to a data location. There is one JIRA ticket open to handle this failure scenario by changing to cmPath which will modify this error flow. That's why here we just continue without caring about failure.

I'll try to fix the double negation.